### PR TITLE
[4.3] Fixing php compatibility issues 3

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
This is one of several PRs to separate out the fixes from #38471. The `mbstring.func_overload` feature was deprectaed in PHP 7.2 and removed in PHP 8.0.

### Testing Instructions
Codereview

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by HLeithner** [comment]
based on the b/c policy, please deprecate and remove with 5.0 pr and add a documentation for this please

**Comment by Hackwar** [comment]
Done. here is the documentation PR: https://github.com/joomla/Manual/pull/54

**Review by wilsonge — CHANGES_REQUESTED** [review]
I think removing the check is wrong for valid PHP Versions (i.e. anything pre-php 8.0 - just because the string is deprecated doesn't mean people may have it set to undesirable values). I think we should just wrap it in a version check statement rather than remove it.

**Comment by HLeithner** [comment]
> I think removing the check is wrong for valid PHP Versions (i.e. anything pre-php 8.0 - just because the string is deprecated doesn't mean people may have it set to undesirable values). I think we should just wrap it in a version check statement rather than remove it.

if it set in a php version that it doesn't support we don't need to care right? But in this case we support php 7.2 which also support mbstring.func_overload right? so I would rebase this pr on 5.0 problem solved.

**Comment by wilsonge** [comment]
That's fine too. But I think @Hackwar wanted to add the phpcs compat checking in now. And if it's removed in 8.0 a version check to not show unnecessary info is fine imo from a minor version

**Comment by HLeithner** [comment]
> That's fine too. But I think @Hackwar wanted to add the phpcs compat checking in now. And if it's removed in 8.0 a version check to not show unnecessary info is fine imo from a minor version

in this case it makes more sense to wrap the code in a phpcs ignore block and add a deprecation notice

File: administrator/language/en-GB/com_joomlaupdate.ini [hunk 1]
```diff
@@ -183,8 +183,10 @@ INSTL_EXTENSION_AVAILABLE="%s Available"
 INSTL_FILE_UPLOADS="File Uploads"
 INSTL_JSON_SUPPORT_AVAILABLE="JSON Support"
 INSTL_MB_LANGUAGE_IS_DEFAULT="MB Language is Default"
+; Deprecated, will be removed with 5.0
```
Comment (wilsonge) [inline administrator/language/en-GB/com_joomlaupdate.ini]: Either remove or change version (same with the rest of the strings)

**Review by wilsonge — CHANGES_REQUESTED** [review]
Minor things but fine

**Comment by HLeithner** [comment]
thanks

**Pull request merged**